tests/test_cases/test_cocotb/test_handle.py

```diff
@@ -433,3 +433,22 @@ async def watch():
     dut.mybits_uninitialized.setimmediatevalue(2)
     await Timer(1, "ns")
     assert seen == 1
+
+
+@cocotb.test(
+    # GHDL uses the VPI, which does not have a way to infer null ranges
+    skip=SIM_NAME.startswith("ghdl"),
+)
+async def test_null_range_width(dut):
+    # Normal arrays should have the same length regardless of language
+    assert len(dut.array_7_downto_4) == 4
+    if LANGUAGE in ["vhdl"]:
+        # But in VHDL, `4 downto 7` should result in a null range
+        try:
+            dut.array_4_downto_7
+        except AttributeError:
+            return
+        assert False, "Expected null array to be deleted"
```

Comment:
Drive-by (full review upcoming once you got it working): Have a look at https://docs.pytest.org/en/stable/how-to/assert.html#assertions-about-expected-exceptions for a nicer way to test an expected exception.